src/tests/ftest/scrubber/performance.py

```diff
@@ -0,0 +1,71 @@
+        :avocado: tags=checksum,scrubber
+        :avocado: tags=TestWithScrubberPerf,test_scrubber_performance
+        """
+        write_flags = self.params.get("write_flags", self.ior_cmd.namespace)
+        read_flags = self.params.get("read_flags", self.ior_cmd.namespace)
+        self.pool = self.get_pool()
+        self.container = self.get_container(self.pool)
+        # First get the write metrics
+        self.ior_cmd.flags.update(write_flags)
+        cmd_result = self.run_ior_with_pool(create_cont=False,
+                                            fail_on_warning=self.log.info)
+        metrics = self.ior_cmd.get_ior_metrics(cmd_result)
+        ior_write_size_without_scrubber = float(metrics[0][IorMetrics.MAX_MIB])
+        # Now, get the read metrics
+        self.ior_cmd.flags.update(read_flags)
+        cmd_result = self.run_ior_with_pool(create_cont=False,
+                                            fail_on_warning=self.log.info)
+        metrics = self.ior_cmd.get_ior_metrics(cmd_result)
+        ior_read_size_without_scrubber = float(metrics[0][IorMetrics.MAX_MIB])
+        self.log.info("IOR Metrics without scrubber enabled")
+        self.log.info("====================================")
+        self.log.info("ior_write_size = %d", ior_write_size_without_scrubber)
+        self.log.info("ior_read_size = %d", ior_read_size_without_scrubber)
+        self.pool.set_prop(properties="scrub:timed,scrub_freq:1")
+        # Wait for the scrubber to scan objects.
+        self.log.info("Waiting for 15 seconds")
+        time.sleep(15)
+        # Get the write metrics with scrubber enabled.
+        self.ior_cmd.flags.update(write_flags)
+        cmd_result = self.run_ior_with_pool(create_cont=False,
+                                            fail_on_warning=self.log.info)
+        metrics = self.ior_cmd.get_ior_metrics(cmd_result)
+        ior_write_size_with_scrubber = float(metrics[0][IorMetrics.MAX_MIB])
+        # Now, get the read metrics with scrubber enabled.
+        self.ior_cmd.flags.update(read_flags)
+        cmd_result = self.run_ior_with_pool(create_cont=False,
+                                            fail_on_warning=self.log.info)
+        metrics = self.ior_cmd.get_ior_metrics(cmd_result)
+        ior_read_size_with_scrubber = float(metrics[0][IorMetrics.MAX_MIB])
+        self.log.info("IOR Metrics with scrubber enabled")
+        self.log.info("=================================")
+        self.log.info("ior_write_size = %d", ior_write_size_with_scrubber)
+        self.log.info("ior_read_size = %d", ior_read_size_with_scrubber)
+        self.assertLessEqual(ior_write_size_with_scrubber,
+                             ior_write_size_without_scrubber,
+                             "Max Write Diff too large")
+        self.assertLessEqual(ior_read_size_with_scrubber,
+                             ior_read_size_without_scrubber,
+                             "Max Read Diff too large")
```

Comment:
I do not understand the purpose of this test.
what is the expected impact of checksum scrubber? what if the scrubber did not introduce any impact while ior is running and the impact was after?
that would mean the <= condition can be false sometimes and the assertion will fail.

Reply:
The test enables scrubber under all conditions by setting scrub_freq:1. This should affect ior performance (atleast that's what we are seeing in the test results). If scrubber is set to lazy mode, it should not affect ior performance and the assert condition happens and the test fails.

Presently, we don't have any baseline information about how much performance is expected on CI clusters. All we are checking here is scrubber should affect performance under certain conditions... We have to run it manually on a performance cluster and get the actual results... This test was initially planned as a manual test case. Based on @daltonbohning input, we are trying to run on weekly testing.

Reply:
> All we are checking here is scrubber should affect performance under certain conditions.

and what im saying is that this check seems worthless to me and just a waste of CI resources

Reply:
@mchaarawi : What is your recommendation? Make this test as manual test? I agree with the CI resources for now.

Reply:
i don't even think this should be a manual test.  is there a test plan you are following where this came from?
it's one thing to measure the affect of scrubber on performance (that should be a manual test and interesting/needed to track); but this is not even doing that.

this is just verifying that scrubber does have an affect on ior performance. which IMO is not always deterministic. there are other factors that play into account here like noise, network issues, that may not have a consistent result.  i don't see a point of doing that..

Reply:
Yes. This test is part of the testing plan : https://daosio.atlassian.net/wiki/spaces/DC/pages/11144036353/Checksum+Scrubbing+Test+Plan

Based on the previous test results are having appx 25 percent degradation due to scrubber (with the present settings in the test case). Please refer ticket: https://daosio.atlassian.net/browse/DAOS-7372

The test provides the IOR bandwidth values with and without scrubber enabled. I am not sure what we have stated in our SOW to the customer. @ryon-jensen : Can you provide your input?

Reply:
historically, CI numbers have not been reliable and this performance eval should be done somewhere on a production system like aurora, lrx, or even frontera (which is not necessary production but can provide meaningful perf results).

Reply:
Yes. I am planning to run this test on Aurora and that was the intention for this manual test creation.

Reply:
i did not say we should avoid running manually. but this test as it is, does not provide any meaningful substance. the manual testing needs to measure the performance effect in percentage of the scrubber and make sure it's reasonable. just verifying there is a performance effect is not necessary IMO.

Reply:
That we can definitely implement... @daltonbohning already stated on one of this comment... I can add the following log info commands:

`dfs_max_write = float(dfs_perf[0][IorMetrics.MAX_MIB]) 
 dfuse_max_write = float(dfuse_perf[0][IorMetrics.MAX_MIB]) 
 actual_write_x = percent_change(dfs_max_write, dfuse_max_write) 
 self.log.info("DFS Max Write:      %.2f", dfs_max_write) 
 self.log.info("DFUSE IL Max Write: %.2f", dfuse_max_write) 
 self.log.info("Percent Diff:       %.2f%%", actual_write_x * 100) 
  
 dfs_max_read = float(dfs_perf[1][IorMetrics.MAX_MIB]) 
 dfuse_max_read = float(dfuse_perf[1][IorMetrics.MAX_MIB]) 
 actual_read_x = percent_change(dfs_max_read, dfuse_max_read) 
 self.log.info("DFS Max Read:       %.2f", dfs_max_read) 
 self.log.info("DFUSE IL Max Read:  %.2f", dfuse_max_read) 
 self.log.info("Percent Diff:       %.2f%%", actual_read_x * 100) `

Reply:
If we want to keep this as a manual test for Aurora, etc. 
We could implement my previous suggestion to enforce a threshold, but as Mohamad pointed out, CI is not reliable:
> historically, CI numbers have not been reliable and this performance eval should be done somewhere on a production system like aurora, lrx, or even frontera (which is not necessary production but can provide meaningful perf results).

So if automating this test in CI doesn't provide value, I think it should be manual. And if it's manual, I don't think we don't need a separate test case. We could run the existing `ftest/performance/*` tests once without scrubber enabled, then update the pool properties in the yaml to enable scrubber and run again.

The trouble with manual tests that don't run in Ci is they always eventually break because it's not possible to verify ftest utility changes don't break them.

Reply:
This I don't have an issue... If everyone is ok, I can close this PR and use ftest/performance scripts and gather the scrubber performance. Yes, we need to update YAML file during testing but I think it should be fine.

Reply:
@daltonbohning  : I am not clear why we are running ftest/performance scripts under CI (if the results are not reliable)? Is it just for sanity check?

Reply:
@rpadma2 Yes, those run in CI so we can verify they aren't broken